src/blueapi/service/main.py

```diff
@@ -87,6 +89,55 @@ def get_state(handler: Handler = Depends(get_handler)) -> WorkerState:
     return handler.worker.state
 
 
+# Map of WorkerState to transition into : allowed prior WorkerStates
+_ALLOWED_TRANSITIONS: Dict[WorkerState, Set[WorkerState]] = {
+    WorkerState.SHUTDOWN: {state for state in WorkerState},
+    WorkerState.RUNNING: {WorkerState.PAUSED},
+    WorkerState.PAUSED: {WorkerState.RUNNING}
+}
+
+
+@app.put("/worker/state", status_code=status.HTTP_400_BAD_REQUEST)
+def set_state(new_state: WorkerState,
+              background_tasks: BackgroundTasks,
+              response: Response,
+              defer: Optional[bool] = False,
+              handler: Handler = Depends(get_handler),
+              ) -> Response:
+    """
+    Request that the worker is put into a particular state.
+    Accepted values:
+    - SHUTDOWN: Attempts to shut down the worker and messaging
+        Returns: 202 "Accepted"
+    - PAUSE: Attempts to pause the worker if it is running
+        query params:
+            defer: Optional[bool] = False
+                If defer is set to true, will wait until the next checkpoint to pause.
+        Returns: 202 "Accepted" if the worker is running, else 400 "Bad Request"
+    - RESUME: Attempts to resume a paused worker
+        Returns: 202 "Accepted" if the worker is paused, else 400 "Bad Request"
+    Rejected values:
+    - All others.
+        Returns: 400 "Bad Request"
+    """
+    current_state = handler.worker.state
+    if new_state in _ALLOWED_TRANSITIONS and current_state in _ALLOWED_TRANSITIONS[new_state]:
```

Comment:
Reversing `_ALLOWED_TRANSITIONS` to be `current -> allowed next state` would let this be reduced to `new_state in ALLOWED[current]`

Reply:
Will flip them over. Currently is only map of PAUSED: RUNNING and RUNNING:PAUSED, so it's as simple as changing the conditional.